lineman/app/models/group_model.coffee

```diff
@@ -148,6 +148,12 @@ angular.module('loomioApp').factory 'GroupModel', (DraftableModel, AppConfig) ->
         @remove()
         _.each @memberships(), (m) -> m.remove()
 
+    markAsRead: =>
+      @remote.postMember(@key, 'mark_as_read')
+      _.map @discussions(), (discussion) ->
+        return unless discussion.isUnread()
+        discussion.update(lastReadAt: moment(), readSalientItemsCount: discussion.salientItemsCount)
```

Comment:
Whoa, now that I'm looking at the code, your suggestion of using discussion_model's mark as read is perfect. There's even that really cool `invoke` function on underscore we can use.